public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/common.ts

```diff
@@ -22,6 +22,21 @@ export const findOptions = (options: AzureMonitorOption[], values: string[] = []
 
 export const toOption = (v: { text: string; value: string }) => ({ value: v.value, label: v.text });
 
+export const addValueToOptions = (
+  values: AzureMonitorOption[],
+  variableOptionGroup: VariableOptionGroup,
+  value?: string
+) => {
+  const options = [...values, variableOptionGroup];
+
+  const optionValues = values.map((m) => m.value.toLowerCase()).concat(variableOptionGroup.options.map((p) => p.value));
```

Comment:
I would do this in one step as:

```
if([...values, variableOptionGroup.options].every((v) => v.value.toLowerCase() !== value.toLowerCase())) {
    options.push({ label: value, value});
}
```

To save on the extra loop the `includes` will add.

Reply:
that doesn't work exactly because options is a because the syntax is not valid:

<img width="1032" alt="Screenshot 2022-12-23 at 18 17 54" src="https://user-images.githubusercontent.com/4025665/209375599-d992cdea-6c17-4a2a-b09d-a2afda28f1fb.png">

making that work (e.g. forcing to read `value` and ensuring it's not `undefined` would make the method less readable. Since these are short arrays, you don't win that much avoiding the extra loop.